### ISSUE TITLE
VideoDescriptionCourseSection not found! StructuredDescriptionPlaylistLockup not found! Type mismatch...

### ISSUE THREAD

**Issue by inad9300** [LuanRT/YouTube.js#493]
### Steps to reproduce

```
InnertubeError: VideoDescriptionCourseSection not found!
This is a bug, want to help us fix it? Follow the instructions at https://github.com/LuanRT/YouTube.js/blob/main/docs/updating-the-parser.md or report it at https://github.com/LuanRT/YouTube.js/issues!
Introspected and JIT generated this class in the meantime:
class VideoDescriptionCourseSection extends YTNode {
  static type = 'VideoDescriptionCourseSection';

  section_title: Text;
  media_lockups: ObservedArray<YTNodes.StructuredDescriptionPlaylistLockup> | null;

  constructor(data: RawNode) {
    super();
    this.section_title = new Text(data.sectionTitle);
    this.media_lockups = Parser.parse(data.mediaLockups, true, [ YTNodes.StructuredDescriptionPlaylistLockup ]);
  }
}

    at ERROR_HANDLER (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:63:26)
    at createRuntimeClass (file:///.../node_modules/youtubei.js/dist/src/parser/generator.js:244:5)
    at generateRuntimeClass (file:///.../node_modules/youtubei.js/dist/src/parser/generator.js:296:21)
    at parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:325:17)
    at Module.parseArray (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:368:28)
    at new StructuredDescriptionContent (file:///.../node_modules/youtubei.js/dist/src/parser/classes/StructuredDescriptionContent.js:11:29)
    at Module.parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:348:28)
    at new EngagementPanelSectionList (file:///.../node_modules/youtubei.js/dist/src/parser/classes/EngagementPanelSectionList.js:12:31)
    at parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:348:28)
    at file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:298:22 {
  date: 2023-08-30T15:18:48.485Z,
  version: '6.2.0'
}
```

```
InnertubeError: StructuredDescriptionPlaylistLockup not found!
This is a bug, want to help us fix it? Follow the instructions at https://github.com/LuanRT/YouTube.js/blob/main/docs/updating-the-parser.md or report it at https://github.com/LuanRT/YouTube.js/issues!
Introspected and JIT generated this class in the meantime:
class StructuredDescriptionPlaylistLockup extends YTNode {
  static type = 'StructuredDescriptionPlaylistLockup';

  thumbnail: Thumbnail[];
  title: Text;
  short_byline_text: Text;
  video_count_short_text: Text;
  endpoint: NavigationEndpoint;
  thumbnail_width: number;
  aspect_ratio: number;
  max_lines_title: number;
  max_lines_short_byline_text: number;
  overlay_position: string;

  constructor(data: RawNode) {
    super();
    this.thumbnail = Thumbnail.fromResponse(data.thumbnail);
    this.title = new Text(data.title);
    this.short_byline_text = new Text(data.shortBylineText);
    this.video_count_short_text = new Text(data.videoCountShortText);
    this.endpoint = new NavigationEndpoint(data.navigationEndpoint);
    this.thumbnail_width = data.thumbnailWidth;
    this.aspect_ratio = data.aspectRatio;
    this.max_lines_title = data.maxLinesTitle;
    this.max_lines_short_byline_text = data.maxLinesShortBylineText;
    this.overlay_position = data.overlayPosition;
  }
}

    at ERROR_HANDLER (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:63:26)
    at createRuntimeClass (file:///.../node_modules/youtubei.js/dist/src/parser/generator.js:244:5)
    at generateRuntimeClass (file:///.../node_modules/youtubei.js/dist/src/parser/generator.js:296:21)
    at generateRuntimeClass (file:///.../node_modules/youtubei.js/dist/src/parser/generator.js:299:9)
    at parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:325:17)
    at Module.parseArray (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:368:28)
    at new StructuredDescriptionContent (file:///.../node_modules/youtubei.js/dist/src/parser/classes/StructuredDescriptionContent.js:11:29)
    at Module.parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:348:28)
    at new EngagementPanelSectionList (file:///.../node_modules/youtubei.js/dist/src/parser/classes/EngagementPanelSectionList.js:12:31)
    at parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:348:28) {
  date: 2023-08-30T15:18:48.489Z,
  version: '6.2.0'
}
```

```
ParsingError: Type mismatch, got VideoDescriptionCourseSection expected VideoDescriptionHeader | ExpandableVideoDescriptionBody | VideoDescriptionMusicSection | VideoDescriptionInfocardsSection | HorizontalCardList.
    at ERROR_HANDLER (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:51:26)
    at parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:329:25)
    at Module.parseArray (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:368:28)
    at new StructuredDescriptionContent (file:///.../node_modules/youtubei.js/dist/src/parser/classes/StructuredDescriptionContent.js:11:29)
    at Module.parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:348:28)
    at new EngagementPanelSectionList (file:///.../node_modules/youtubei.js/dist/src/parser/classes/EngagementPanelSectionList.js:12:31)
    at parseItem (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:348:28)
    at file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:298:22
    at Array.map (<anonymous>)
    at Module.parseResponse (file:///.../node_modules/youtubei.js/dist/src/parser/parser.js:297:100) {
  info: { sectionTitle: { runs: [Array] }, mediaLockups: [ [Object] ] },
  date: 2023-08-30T15:18:48.489Z,
  version: '6.2.0'
}
```

```
error: ViewCountFactoid not found!
This is a bug, want to help us fix it? Follow the instructions at https://github.com/LuanRT/YouTube.js/blob/main/docs/updating-the-parser.md or report it at https://github.com/LuanRT/YouTube.js/issues!
Introspected and JIT generated this class in the meantime:
class ViewCountFactoid extends YTNode {
  static type = 'ViewCountFactoid';

  view_count_entity_key: string;
  factoid: YTNodes.Factoid | null;
  view_count_type: string;

  constructor(data: RawNode) {
    super();
    this.view_count_entity_key = data.viewCountEntityKey;
    this.factoid = Parser.parseItem(data.factoid, [ YTNodes.Factoid ]);
    this.view_count_type = data.viewCountType;
  }
}

      at new InnertubeError (.../node_modules/youtubei.js/dist/src/utils/Utils.js:21:8)
      at ERROR_HANDLER (.../node_modules/youtubei.js/dist/src/parser/parser.js:63:25)
      at createRuntimeClass (.../node_modules/youtubei.js/dist/src/parser/generator.js:244:4)
      at generateRuntimeClass (.../node_modules/youtubei.js/dist/src/parser/generator.js:296:20)
      at parseItem (.../node_modules/youtubei.js/dist/src/parser/parser.js:325:16)
      at parseArray (.../node_modules/youtubei.js/dist/src/parser/parser.js:368:27)
      at new VideoDescriptionHeader (.../node_modules/youtubei.js/dist/src/parser/classes/VideoDescriptionHeader.js:15:24)
      at parseItem (.../node_modules/youtubei.js/dist/src/parser/parser.js:348:27)
      at parseArray (.../node_modules/youtubei.js/dist/src/parser/parser.js:368:27)
      at new StructuredDescriptionContent (.../node_modules/youtubei.js/dist/src/parser/classes/StructuredDescriptionContent.js:11:21)

16 | }
17 | _a = Platform;
18 | _Platform_shim = { value: void 0 };
19 | export class InnertubeError extends Error {
20 |     constructor(message, info) {
21 |         super(message);
            ^
error: Type mismatch, got ViewCountFactoid expected Factoid.
 info: {
  "viewCountEntityKey": "Egt1bXg2M3lNSkptbyDCASgB",
  "factoid": {
    "factoidRenderer": {
      "value": {
        "simpleText": "797"
      },
      "label": {
        "simpleText": "Views"
      },
      "accessibilityText": "797 views"
    }
  },
  "viewCountType": "VIEW_COUNT_FACTOID_TYPE_TOTAL_VIEWS"
}

      at new InnertubeError (.../node_modules/youtubei.js/dist/src/utils/Utils.js:21:8)
      at new ParsingError (:1:32)
      at ERROR_HANDLER (.../node_modules/youtubei.js/dist/src/parser/parser.js:51:25)
      at parseItem (.../node_modules/youtubei.js/dist/src/parser/parser.js:339:20)
      at parseArray (.../node_modules/youtubei.js/dist/src/parser/parser.js:368:27)
      at new VideoDescriptionHeader (.../node_modules/youtubei.js/dist/src/parser/classes/VideoDescriptionHeader.js:15:24)
      at parseItem (.../node_modules/youtubei.js/dist/src/parser/parser.js:348:27)
      at parseArray (.../node_modules/youtubei.js/dist/src/parser/parser.js:368:27)
      at new StructuredDescriptionContent (.../node_modules/youtubei.js/dist/src/parser/classes/StructuredDescriptionContent.js:11:21)
      at parseItem (.../node_modules/youtubei.js/dist/src/parser/parser.js:348:27)
```

### Version

Default

### Checklist

- [X] I am running the latest version.
- [X] I checked the documentation and found no answer.
- [X] I have searched the existing issues and made sure this is not a duplicate.
- [X] I have provided sufficient information.

**Comment by LuanRT** [comment]
Hi there, thanks for reporting this! Is there a specific video id we can use to reproduce this?

**Comment by inad9300** [comment]
`wTWd5SwQILg`, `-GyLWrDF8ss`, `CwXixVcliP0`, `3PGbzLCfg6Y` and many others from the same channel.

**Comment by inad9300** [comment]
I seem to be getting new errors by the day. It is a rare occurrence that I attempt to process a channel's videos and it goes smoothly, to be honest. I know how much a pain in the ass extracting data from YouTube can be, but I wanted to voice this observation nevertheless.
 
```
📥 https://youtu.be/l_qY2p0OH9A
16 | }
17 | _a = Platform;
18 | _Platform_shim = { value: void 0 };
19 | export class InnertubeError extends Error {
20 |     constructor(message, info) {
21 |         super(message);
            ^
error: Type mismatch, got VideoDescriptionTranscriptSection expected VideoDescriptionHeader | ExpandableVideoDescriptionBody | VideoDescriptionMusicSection | VideoDescriptionInfocardsSection | HorizontalCardList.
 info: {
  "sectionTitle": {
    "runs": [
      {
        "text": "Transcript"
      }
    ]
  },
  "subHeaderText": {
    "runs": [
      {
        "text": "Follow along using the transcript."
      }
    ]
  },
  "primaryButton": {
    "buttonRenderer": {
      "style": "STYLE_SUGGESTIVE",
      "size": "SIZE_DEFAULT",
      "isDisabled": false,
      "text": {
        "runs": [
          {
            "text": "Show transcript"
          }
        ]
      },
      "trackingParams": "CCIQtooIIhMIk6e3jdmogQMVqtpJBx1B1AWi",
      "command": {
        "clickTrackingParams": "CCIQtooIIhMIk6e3jdmogQMVqtpJBx1B1AWi",
        "commandExecutorCommand": {
          "commands": [
            {
              "clickTrackingParams": "CCIQtooIIhMIk6e3jdmogQMVqtpJBx1B1AWi",
              "showEngagementPanelEndpoint": {
                "panelIdentifier": "engagement-panel-searchable-transcript",
                "sourcePanelIdentifier": "video-description-ep-identifier"
              }
            },
            {
              "clickTrackingParams": "CCIQtooIIhMIk6e3jdmogQMVqtpJBx1B1AWi",
              "scrollToEngagementPanelCommand": {
                "targetId": "engagement-panel-searchable-transcript"
              }
            }
          ]
        }
      }
    }
  },
  "trackingParams": "CCEQ0YQIGAMiEwiTp7eN2aiBAxWq2kkHHUHUBaI="
}

      at new InnertubeError (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/utils/Utils.js:21:8)
      at new ParsingError (:1:32)
      at ERROR_HANDLER (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/parser.js:52:25)
      at parseItem (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/parser.js:327:24)
      at parseArray (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/parser.js:366:27)
      at new StructuredDescriptionContent (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/classes/StructuredDescriptionContent.js:11:21)
      at parseItem (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/parser.js:346:27)
      at new EngagementPanelSectionList (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/classes/EngagementPanelSectionList.js:13:23)
      at parseItem (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/parser.js:346:27)
      at parseArray (/Users/danielma/Projects/captain-caption/backend/node_modules/youtubei.js/dist/src/parser/parser.js:366:27)

💥 Cannot access uninitialized variable.
```

**Comment by LuanRT** [comment]
All these warnings/errors have been fixed by #511 and two other PRs I merged a few weeks ago. Consider updating to the latest version once it's released.
